spring-kafka/src/main/java/org/springframework/kafka/support/converter/BatchMessagingMessageConverter.java

```diff
@@ -106,6 +140,20 @@ public void setGenerateTimestamp(boolean generateTimestamp) {
 			offsets.add(record.offset());
 			timestampTypes.add(record.timestampType().name());
 			timestamps.add(record.timestamp());
+			if (this.headerMapper != null) {
+				Map<String, Object> converted = new HashMap<>();
+				this.headerMapper.toHeaders(record.headers(), converted);
+				convertedHeaders.add(converted);
+			}
+			else {
+				if (record.headers().iterator().hasNext() && logger.isWarnEnabled()) {
```

Comment:
So, we see the WARN only if we have entries in the `record.headers()`, but at the same time we populate the `KafkaHeaders.NATIVE_HEADERS` anyway, independently of the content.

Does it make sense to restrict warn then?